src/utils/pgwire/src/pg_protocol.rs

```diff
@@ -363,14 +363,19 @@ where
         let result = self.inner_process_query_msg(sql, session).await;
 
         let mills = start.elapsed().as_millis();
-        let truncated_sql = &sql[..std::cmp::min(sql.len(), 1024)];
+        let truncated = sql.len() > 1024;
+
         tracing::info!(
             target: PGWIRE_QUERY_LOG,
             mode = %"(simple query)",
             session = %session_id,
             status = %if result.is_ok() { "ok" } else { "err" },
-            time = %format!("{}ms", mills),
-            sql = %truncated_sql,
+            time = %format_args!("{}ms", mills),
+            sql = %format_args!(
+                "{}{}",
+                if truncated { &sql[..1024] } else { sql },
+                if truncated { "...(truncated)" } else { "" }
+            )
```

Comment:
Oh, this place used a different implementation by chance. Just right for a comparison. 🤔

Reply:
Either LGTM